tests/Unit/Framework/TestHelpers.hpp

```diff
@@ -395,6 +395,47 @@ void test_throw_exception(const ThrowingFunctor& func,
                         << __LINE__;                                    \
   }())
 
+/*!
+ * \ingroup TestingFrameworkGroup
+ * \brief A random element between `start` and `end`
+ */
+template <typename Iter>
+Iter random_sample(Iter start, Iter end,
```

Comment:
This seems pretty generic, maybe move to `Utilities/RandomSample.hpp`?

Reply:
I can't think of a use case outside of tests right now, so I suggest we keep it in `tests/` for now